Mongo db Interface with Unit Tests

seanlangbrown opened the pull request:
MongoDB interface with unit tests using mongoDB-memory-server.  Working on circleCI!

Possible issues with the babel configuration (had errors on earlier commits from using some es6 syntax on circleCI), please comment on the package.json if you can.

Rebase required to merge because I branched before merging the previous PR.

Thanks!

seanlangbrown committed:
implement mongoDB interface unsing mongoose, mock review data and seed worker

seanlangbrown committed:
implement jest test for db interfaces using mongodb-memory-server

seanlangbrown committed:
add mongodb-memory-server and lodash to dependencies

seanlangbrown committed:
debug db interface

seanlangbrown committed:
debug mockData

seanlangbrown committed:
complete database interface and unit tests using in-memory mongodb

seanlangbrown committed:
implement seedDB using async/await

seanlangbrown committed:
update package.json to use babel for transpiling es2015

seanlangbrown committed:
re-add mongodb-memory-server to dependencies

seanlangbrown committed:
debug db interface in strict mode

seanlangbrown committed:
debug db index.js - replace ... operator to avoid error in circleCI

seanlangbrown committed:
update package.json jest configuration with newer syntax

seanlangbrown committed:
debug jest syntax in package.json

seanlangbrown committed:
adjust jest setTimeout to allow for download of MongoDBMemoryServer beforeAll in db index.test

seanlangbrown committed:
add debugg logging statements to db index.js

seanlangbrown committed:
debug mockData is undefined in db index.js on circleCI

seanlangbrown committed:
debug It should add experiences to the db timeout on circleCI

seanlangbrown committed:
refactor assertions in It should add experiences to the db to debug timeout on CircleCI

seanlangbrown committed:
increase timeout to debug jest timeout on circle ci

seanlangbrown committed:
complete db/index.test.js remove debugging console logs

seanlangbrown committed:
experiement with removing before all initialization on in memory mongoDB

ischulz commented:
So I just looked through I believe all of them. It looks good and great you got it to work. I wonder though why do you require mongoose as mongoose in the test and as mongo in the index. I dont think it matters  and maybe it caused a bug. I was just confused for a second.

In your package.json the "unpackedModulePathPattern" I do not know what that is.

seanlangbrown committed:
fix jest test dependencies

seanlangbrown committed:
clean up tests and package.json in response to PR reviews

seanlangbrown commented:
Thanks for the comments Ilias.  
I actually don't need to require mongoose in both the test and index files, so I removed that from the test file.
As for the "unpackedModulePathPattern", I don't know what that does either.  I added it when I was trying to set up transpilation from ES2015 for the Jest tests on circleCI.  Removing it doesn't seem to affect anything, so I've taken it out.  I think it may have something to do with transpiling react.  I was using this link: http://www.hackingwithreact.com/read/1/32/how-to-configure-jest-to-test-react-and-es6
